pandas/core/reshape/merge.py

```diff
@@ -2392,6 +2397,15 @@ def _factorize_keys(
         assert isinstance(rk, BaseMaskedArray)
         llab = rizer.factorize(lk._data, mask=lk._mask)
         rlab = rizer.factorize(rk._data, mask=rk._mask)
+    elif isinstance(lk, ArrowExtensionArray) or isinstance(rk, ArrowExtensionArray):
+        # we can only get here with numeric dtypes
+        # TODO: Remove when we have a Factorizer for Arrow
+        llab = rizer.factorize(
+            lk.to_numpy(na_value=1, dtype=lk.dtype.numpy_dtype), mask=lk.isna()
```

Comment:
Curious why `1` is a safe sentinel for NA vals

Reply:
Works for all numeric dtypes (as long as I am not missing anything)

Reply:
What if we're joining on numeric keys like `lk = [1, NA], `rk = [1, NA, 2]`?

Reply:
That's what the mask is for?

Reply:
Sorry I might be missing something, but we are effectively treating them as our own masked arrays here and missing values are handled by the mask there as well while the value used as placeholder is irrelevant

Reply:
Ah okay I didn't know merging of the NAs are masked away.